auth/api/iam/api.go

```diff
@@ -373,6 +344,50 @@ func (r Wrapper) HandleAuthorizeRequest(ctx context.Context, request HandleAutho
 	}
 }
 
+// GetRequestJWT returns the Request Object referenced as 'request_uri' in an authorization request.
+// RFC9101: The OAuth 2.0 Authorization Framework: JWT-Secured Authorization Request (JAR).
+func (r Wrapper) GetRequestJWT(ctx context.Context, request GetRequestJWTRequestObject) (GetRequestJWTResponseObject, error) {
+	signedRequestObject := new([]byte)
+	err := r.authzRequestObjectStore().Get(request.Id, signedRequestObject)
+	if err != nil {
+		return nil, err
+	}
+	return GetRequestJWT200ApplicationoauthAuthzReqJwtResponse{
+		Body:          bytes.NewReader(*signedRequestObject),
+		ContentLength: int64(len(*signedRequestObject)),
+	}, nil
+}
+
+// PostRequestJWT returns the Request Object referenced as 'request_uri' in an authorization request.
+// Extension of OpenID 4 Verifiable Presentations (OpenID4VP) on
+// RFC9101: The OAuth 2.0 Authorization Framework: JWT-Secured Authorization Request (JAR).
+func (r Wrapper) PostRequestJWT(ctx context.Context, request PostRequestJWTRequestObject) (PostRequestJWTResponseObject, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (r Wrapper) parseJARRequest(ctx context.Context, q url.Values) (oauthParameters, error) {
+	var rawRequestObject string
+	var err error
+	if q.Has(oauth.RequestParam) {
+		if q.Has(oauth.RequestURIParam) {
+			return nil, oauth.OAuth2Error{Code: oauth.InvalidRequest, Description: "claims 'request' and 'request_uri' are mutually exclusive"}
+		}
+		rawRequestObject = q.Get(oauth.RequestParam)
+	} else if q.Has(oauth.RequestURIParam) {
+		requestURI := q.Get(oauth.RequestURIParam)
+		rawRequestObject, err = r.auth.IAMClient().RequestObject(ctx, requestURI)
+		if err != nil {
+			return nil, oauth.OAuth2Error{Code: oauth.InvalidRequestURI, Description: "failed to get Request Object", InternalError: err}
+		}
+	} else {
+		// require_signed_request_object is true, so we reject anything that isn't
+		return nil, oauth.OAuth2Error{Code: oauth.InvalidRequest, Description: "authorization request are required to use signed request objects (RFC9101)"}
+	}
+
+	// already OAuth errors
+	return r.validateJARRequest(ctx, rawRequestObject, q.Get(oauth.ClientIDParam))
```

Comment:
shouldn't there be an existence check on the `oauth.ClientIDParam` parameter?

Reply:
`validateJARRequest` compares the value to the singerKID, so will fail if missing or empty